source/reference/mongotranslate.txt

```diff
@@ -0,0 +1,123 @@
+==================
+``mongotranslate``
+==================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. only:: html
+
+   .. meta::
+      :description: The mongotranslate reference page.
+      :keywords: mongotranslate, mongodb, man page
+
+Description
+-----------
+
+.. only:: (not man)
+
+   .. binary:: mongotranslate
+
+``mongotranslate`` takes a specified SQL query and returns
+an equivalent MongoDB :manual:`aggregation pipeline </aggregation>`. It
+is designed to help users understand how |bi-short| translates
+SQL queries for use with MongoDB databases.
+
+``mongotranslate`` is a standalone program. It requires
+a :doc:`drdl </reference/mongodrdl>` file generated by
+:binary:`~bin.mongodrdl`, but does not require
+:binary:`~bin.mongosqld` or :binary:`~bin.mongod` to be running at the
+time of execution.
+
+Usage
+-----
+
+``mongotranslate`` has the following syntax:
+
+.. code-block:: javascript
+
+   mongotranslate [--query | --queryFile] <query | queryfile> [options]
+
+Command Line Options
+--------------------
+
+.. program:: mongotranslate
+
+.. include:: /includes/option/option-mongotranslate-query.rst
+
+.. include:: /includes/option/option-mongotranslate-queryFile.rst
+
+.. include:: /includes/option/option-mongotranslate-schema.rst
+
+.. include:: /includes/option/option-mongotranslate-dbName.rst
+
+.. include:: /includes/option/option-mongotranslate-format.rst
+
+.. include:: /includes/option/option-mongotranslate-explain.rst
+
+.. include:: /includes/option/option-mongotranslate-mongoversion.rst
+
+Examples
+--------
+
+The following example specifies an inline query for translation:
+
+.. code-block:: javascript
+
+   mongotranslate --query=“select test.name from restaurants where name like 'Brooklyn%'” \
+   --schema=schema.drdl
+
+The above command returns the following results:
+
+.. code-block:: json
+   :copyable: false
+
+   [
+     {"$match": {"name": {"$regex": "^Brooklyn.*$","$options": "i"}}},
+     {"$project": {"test_DOT_restaurants_DOT_name": "$name","_id": {"$numberInt":"0"}}},
+   ]
+
+.. note::
+
+   If you need to quote a string inside the :option:`--query
+   <mongotranslate --query>` parameter, be sure to use single quotes.
+
+The following example specifies a query file and returns a one-line
+result:
+
+.. code-block:: javascript
+
+   mongotranslate --queryFile=query.txt --schema=schema.drdl --format=none
+
+The following example uses the :option:`--explain` option:
+
+.. code-block:: javascript
+
+   mongotranslate --query="select count(name) from restaurants;" \
+   --schema=schema.drdl --explain
+
+The above command returns the following results:
```

Comment:
Please discuss in tech review with @rychipman 

Should we have a section above "Examples" that provides descriptions on the fields in the output? These fields don't look like the explain output in the manual, specifically "PushdownFailures", which I thought was a BIC term.

Perhaps we could provide a warning that seeing those might degrade performance.

Reply:
which manual are you referring to? I couldn't find a manual page for `EXPLAIN` on the docs site or the staged pages. the output here is a json-formatted version of the bic's `EXPLAIN <query>`

Reply:
I think there's some confusion here about the explain output. @jdestefano-mongo you're right that the PushdownFailures are BIC specific - that's because the explain output here is different from MongoDB's explain output. @rychipman I think he's referring to the docs for MongoDB `explain`, which right now are linked on this page in the `--explain` section (I added another comment asking that we remove this). But yes because this `explain` is different from the one that's already documented, we should definitely add more detail about what it is and what each field means!